src/controller_examples/rabbitmq_controller/proof/safety.rs

```diff
@@ -211,34 +211,120 @@ pub proof fn lemma_always_at_most_one_create_cm_req_since_rest_id_is_in_flight(
             always(lift_state(at_most_one_create_cm_req_since_rest_id_is_in_flight(key, rest_id)))
         ),
 {
-    lemma_always_filtered_create_cm_req_len_is_at_most_one(spec, key, rest_id);
-
-    let p = lift_state(at_most_one_create_cm_req_since_rest_id_is_in_flight(key, rest_id));
-    let q = lift_state(filtered_create_cm_req_len_is_at_most_one(key, rest_id));
+    let init = |s: ClusterState| {
```

Comment:
Actually I was about to say that we should refactor these lemmas in this way. Glad that you are already doing so.